src/lazy/text/buffer.rs

```diff
@@ -3046,6 +3057,22 @@ mod tests {
         ]
     }
 
+    matcher_tests_with_macro! {
+        allow_omitting_trailing_optionals
+        match_e_expression
+        "(macro foo (a b+ c? d*) null)"
+        expect_match: [
+            "(:foo 1 2)",
+            "(:foo 1 2 3)",
+            "(:foo 1 2 3 4)",
+            "(:foo 1 2 3 4 5 6)", // implicit rest
```

Comment:
```suggestion
            "(:foo 1 2 3 4 5 6)", // implicit rest
            "(:foo 1 2 (: 3) 4 5 6)", // implicit rest
            "(:foo 1 2 3 (: 4 5 6))",
```
...unless these are already covered elsewhere.